### PR TITLE
Use integer abs function where necessary

### PR DESCRIPTION
**Pull request by selaux**
Follow up of  #388 i replaced all `abs()` that were not explicitly `fabs()` with a macro that does a integer abs.

This seems to fix (at least) the AI deadlocks occuring during enemy turns.

### CODE REVIEW

File: sgp/Types.h [hunk 1]
```diff
@@ -24,6 +24,7 @@
 #define endof(a) ((a) + lengthof(a))
 
 
+#define ABS(a) (UINT16)abs((int)a)
```
Comment (lynxlynxlynx) [inline sgp/Types.h]: Should be (int)(a) or simply int(a) so the result is cast, not only the first argument in non-trivial expressions.

And while the change seems automated, more whitespace errors crept up.

Comment (selaux) [inline sgp/Types.h]: Should both be fixed. Thanks for the review.

File: Build/TileEngine/Lighting.cc [hunk 2]
```diff
@@ -537,9 +537,9 @@ static DOUBLE LinearDistanceDouble(INT16 iX1, INT16 iY1, INT16 iX2, INT16 iY2)
 {
 INT32 iDx, iDy;
```
Comment (derSteFfi) [inline Build/TileEngine/Lighting.cc]: maybe we should make those unsigned... the parsing might give errors.

Comment (derSteFfi) [inline Build/TileEngine/Lighting.cc]: of course not really part of this PR
